src/device.cpp

```diff
@@ -204,28 +204,21 @@ NAN_METHOD(GetDeviceInfo) {
     return;
   }
   // cl_ulong params
-  case CL_DEVICE_GLOBAL_MEM_CACHE_SIZE:
-  case CL_DEVICE_GLOBAL_MEM_SIZE:
   case CL_DEVICE_LOCAL_MEM_SIZE:
+  case CL_DEVICE_GLOBAL_MEM_SIZE:
+  case CL_DEVICE_MAX_MEM_ALLOC_SIZE:
+  case CL_DEVICE_GLOBAL_MEM_CACHE_SIZE:
   case CL_DEVICE_MAX_CONSTANT_BUFFER_SIZE:
-  case CL_DEVICE_MAX_MEM_ALLOC_SIZE: {
+  {
     cl_ulong param_value;
     CHECK_ERR(::clGetDeviceInfo(device_id, param_name, sizeof(cl_ulong), &param_value, NULL));
 
     /**
       JS Compatibility
 
-      As JS does not support 64 bits integer, we return a 2-integer array with
-        output_values[0] = (input_value >> 32) & 0xffffffff;
-        output_values[1] = input_value & 0xffffffff;
-
-      and reconstruction as
-        input_value = ((int64_t) output_values[0]) << 32) | output_values[1];
+      As JS does not support 64 bits integer, we return the amount as a 32 bit value containing the amount in kilobytes, it should be enough for couple of generations of GPUs, in future the value can be returned in megabytes / gigabytes / etc. with library's major version bump.
     */
-    Local<Array> arr = Nan::New<Array>(2);
-    Nan::Set(arr, 0, JS_INT((uint32_t) (param_value>>32))); // hi
-    Nan::Set(arr, 1, JS_INT((uint32_t) (param_value & 0xffffffff))); // lo
-    info.GetReturnValue().Set(arr);
+    info.GetReturnValue().Set(JS_INT((uint32_t) (param_value >> 10)));
```

Comment:
Changing the unit means this API doesn't conform to the OpenCL spec, and I worry the left shift will lead to precision loss on certain devices. What's wrong with returning the long as a pair of lo,hi 32-bit numbers? They can very easily be turned into a bigint in JS now days.